be/src/exprs/string_functions.cpp

```diff
@@ -761,4 +761,63 @@ StringVal StringFunctions::money_format(FunctionContext *context, const LargeInt
     return do_money_format(context, ss.str());
 }
 
+static int index_of(const uint8_t* source, int sourceOffset, int sourceCount,
+                const uint8_t* target, int targetOffset, int targetCount,
+                int fromIndex) {
+    if (fromIndex >= sourceCount) {
+        return (targetCount == 0 ? sourceCount : -1);
+    }
+    if (fromIndex < 0) {
+        fromIndex = 0;
+    }
+    if (targetCount == 0) {
+        return fromIndex;
+    }
+    const uint8_t first = target[targetOffset];
+    int max = sourceOffset + (sourceCount - targetCount);
+    for (int i = sourceOffset + fromIndex; i <= max; i++) {
+        if (source[i] != first) { // Look for first character
+            while (++i <= max && source[i] != first);
+        }
+        if (i <= max) { // Found first character, now look at the rest of v2
+            int j = i + 1;
+            int end = j + targetCount - 1;
+            for (int k = targetOffset + 1; j < end && source[j] == target[k]; j++, k++);
+            if (j == end) {
+                return i - sourceOffset; // Found whole string.
+            }
+        }
+    }
+    return -1;
+}
+
+
+StringVal StringFunctions::split_part(FunctionContext* context, const StringVal& content,
+                                      const StringVal& delimiter, const IntVal& field) {
+    if (content.is_null || delimiter.is_null || field.is_null || field.val <= 0) {
+        return StringVal::null();
+    }
+    int find[field.val]; //store substring position
```

Comment:
use std::vector<int> to avoid stack overflow
```suggestion
    std::vector<int> find(field.val, -1); //store substring position
```